test/source/tests/tests/settings.ts

```diff
@@ -143,6 +143,28 @@ export let defineSettingsTests = (testVariant: TestVariant, testWithBrowser: Tes
       internalTestState.expectiIntentionalErrReport = true;
     }));
 
+    ava.default('settings - attachment previews are rendered according to their types', testWithBrowser('compatibility', async (t, browser) => {
+      const inboxPage = await browser.newPage(t, TestUrls.extension(`chrome/settings/inbox/inbox.htm?acctEmail=flowcrypt.compatibility@gmail.com&threadId=1729eee9493eb76a`));
+      // image
+      const attachmentImage = await inboxPage.getFrame(['attachment.htm', 'name=tiny-face.png']);
+      await attachmentImage.click('body');
+      const attachmentPreviewImage = await inboxPage.getFrame(['attachment_preview.htm']);
+      await attachmentPreviewImage.waitAll('#attachment-preview-container img.attachment-preview-img');
+      await inboxPage.press('Escape');
+      // text
+      const attachmentText = await inboxPage.getFrame(['attachment.htm', 'name=small.txt']);
+      await attachmentText.click('body');
+      const attachmentPreviewText = await inboxPage.getFrame(['attachment_preview.htm']);
+      await attachmentPreviewText.waitForContent('#attachment-preview-container .attachment-preview-txt', 'small text file');
+      await inboxPage.press('Escape');
+      // no preview
```

Comment:
Would it eventually be possible to preview pdfs? :) browsers should be able to do it

Reply:
> Would it eventually be possible to preview pdfs? :) browsers should be able to do it

Yes, but I didn't manage to perform pdf preview with native ways (iframe/embed/object) so the solution would probably require 3rd-party solution like Mozilla's [pdf.js](https://github.com/mozilla/pdf.js)